### ISSUE TITLE
tsh login to another cluster fails after relogin

### ISSUE THREAD

**Issue by Joerger** [gravitational/teleport#6268]
### Description
`tsh login` refreshes the public key and the ssh certificate of the profile's current cluster, but not the ssh certificate of other clusters. This leads to an error: `public key in profile does not match the public key in SSH certificate` due to the public keys for the certificate and the login being unequal.

Error location: https://github.com/gravitational/teleport/blob/master/lib/client/interfaces.go#L379-L381. 

Removing the above linked check added in #6213 removes this error.

**What happened**:
```bash
$ ./tsh.sh login --insecure cluster1
ERROR: public key in profile does not match the public key in SSH certificate
```

**What you expected to happen**:
```bash
$ ./tsh.sh login --insecure cluster1
# Successfully login
```

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Setup a trusted cluster
2. Login to the root's proxy
3. Login to the leaf cluster (leaf cluster becomes the profile's current cluster)
4. Re-login to the root's proxy
5. Try to login to the root cluster - fails.

### Server Details
- Teleport version (run `teleport version`): branch/v6
- 
### Debug Logs
Nothing of note

**Comment by awly** [comment]
@Joerger can you try with https://github.com/gravitational/teleport/pull/6251 ?

**Comment by Joerger** [comment]
Yes Just saw that PR.

**Comment by Joerger** [comment]
Fixed by #6251
